src/plugins/skills/crafting/spinning-wheel-plugin.ts

```diff
@@ -0,0 +1,197 @@
+    [98, {shouldTakeInput: false, count: 10, spinnable: ballOfWool}],
+    [97, {shouldTakeInput: true, count: 0, spinnable: ballOfWool}],
+    [95, {shouldTakeInput: false, count: 1, spinnable: bowString}],
+    [94, {shouldTakeInput: false, count: 5, spinnable: bowString}],
+    [93, {shouldTakeInput: false, count: 10, spinnable: bowString}],
+    [91, {shouldTakeInput: true, count: 0, spinnable: bowString}],
+    [107, {shouldTakeInput: false, count: 1, spinnable: magicAmuletString}],
+    [106, {shouldTakeInput: false, count: 5, spinnable: magicAmuletString}],
+    [105, {shouldTakeInput: false, count: 10, spinnable: magicAmuletString}],
+    [104, {shouldTakeInput: true, count: 0, spinnable: magicAmuletString}],
+    [121, {shouldTakeInput: false, count: 1, spinnable: rootsCbowString}],
+    [120, {shouldTakeInput: false, count: 5, spinnable: rootsCbowString}],
+    [119, {shouldTakeInput: false, count: 10, spinnable: rootsCbowString}],
+    [118, {shouldTakeInput: true, count: 0, spinnable: rootsCbowString}],
+    [114, {shouldTakeInput: false, count: 1, spinnable: sinewCbowString}],
+    [113, {shouldTakeInput: false, count: 5, spinnable: sinewCbowString}],
+    [112, {shouldTakeInput: false, count: 10, spinnable: sinewCbowString}],
+    [111, {shouldTakeInput: true, count: 0, spinnable: sinewCbowString}],
+]);
+
+export const openSpinningInterface: objectAction = (details) => {
+    details.player.activeWidget = {
+        widgetId: widgets.whatWouldYouLikeToSpin,
+        type: 'SCREEN',
+        closeOnWalk: true
+    };
+};
+
+const spinProduct: any = (details: ButtonActionDetails, spinnable: Spinnable, count: number) => {
+    let elapsedTicks = 0;
+
+    let created = 0;
+
+    // As an multiple items can be used for one of the recipes, check if its an array
+    let currentItem: number;
+    let currentItemIndex: number = 0;
+    let isArray = false;
+    if (Array.isArray(spinnable.input)) {
+        isArray = true;
+        currentItem = spinnable.input[0];
+    } else {
+        currentItem = spinnable.input;
+    }
+    // Create a new tick loop
+    const loop = loopingAction(details.player);
+    loop.event.subscribe(() => {
+        if (created === count) {
+            loop.cancel();
+            return;
+        }
+        // Check if out of input material
+        if (!details.player.hasItemInInventory(currentItem)) {
+            let cancel = false;
+            if (isArray) {
+                if (currentItemIndex < (<number[]> spinnable.input).length) {
+                    currentItemIndex++;
+                    currentItem = (<number[]> spinnable.input)[currentItemIndex];
+                } else {
+                    cancel = true;
+                }
+            } else {
+                cancel = true;
+            }
+            if (cancel) {
+                details.player.outgoingPackets.chatboxMessage(`You don't have any ${gameCache.itemDefinitions.get(currentItem).name.toLowerCase()}.`);
```

Comment:
`player.sendMessage(message, true)`

Reply:
for this one its just sent in the chatbox in osrs.